qiskit/circuit/library/__init__.py

```diff
@@ -208,42 +318,79 @@
    UCRYGate
    UCRZGate
 
+The above objects derive :class:`.Gate` or :class:`~.circuit.Instruction`, which allows the
+compiler to reason about them on an abstract level. We therefore suggest using these instead
+of the following which derive :class:`.QuantumCircuit` and are eagerly constructed.
+
+.. autosummary::
+   :toctree: ../stubs/
+   :template: autosummary/class_no_inherited_members.rst
+
+   Diagonal
+   MCMT
+   MCMTVChain
+   Permutation
+   GMS
+   GR
+   GRX
+   GRY
+   GRZ
+
+.. _boolean-logic:
+
 Boolean Logic Circuits
 ======================
 
-These are :class:`~qiskit.circuit.QuantumCircuit` subclasses
-that implement boolean logic operations, such as the logical
+These :class:`.Gate`\ s implement boolean logic operations, such as the logical
 or of a set of qubit states.
 
+.. autosummary::
+   :toctree: ../stubs/
+
+   AndGate
+   OrGate
+   BitwiseXorGate
+   random_bitwise_xor
```

Comment:
This is technically not a class (as mentioned above), could be singled out as a builder function.

Reply:
I moved it out to a dedicated paragraph 👍🏻